queries/smali/highlights.scm

```diff
@@ -4,52 +4,106 @@
 
 (primitive_type) @type.builtin
 
-(array_type
-  "[" @punctuation.bracket)
+((class_identifier) @type.builtin
+  (#lua-match? @type.builtin "^Landroid/"))
+((class_identifier) @type.builtin
+  (#lua-match? @type.builtin "^Lcom/android/"))
+((class_identifier) @type.builtin
+  (#lua-match? @type.builtin "^Ldalvik/"))
+((class_identifier) @type.builtin
+  (#lua-match? @type.builtin "^Ljava/"))
 
 ; Methods
 
-(method_declaration
-  (method_identifier) @method)
+(method_definition
+  (method_signature (method_identifier) @method))
 
-(statement
+(expression
   (opcode) @_invoke
-  (full_method_identifier
-    (method_identifier) @method.call)
+  (value
+	(body
+	  (full_method_signature
+        (method_signature (method_identifier) @method.call))))
   (#lua-match? @_invoke "^invoke"))
 
-(method_identifier
-  "<init>(" @constructor
-  ")" @constructor)
+(method_handle
+  (full_method_signature
+	(method_signature (method_identifier) @method.call)))
+
+(call_site) @method.call
+
+(custom_invoke
+  (method_signature (method_identifier) @method.call))
+
+(annotation_value
+  (body
+    (method_signature (method_identifier) @method.call)))
+
+(annotation_value
+  (body
+    (full_method_signature
+      (method_signature (method_identifier) @method.call))))
+
+(field_definition
+  (value
+	(body
+		(method_signature (method_identifier) @method.call))))
 
-(method_identifier
-  "<clinit>(" @constructor
-  ")" @constructor)
+(field_definition
+  (value
+	(body
+	  (full_method_signature
+		(method_signature (method_identifier) @method.call)))))
+
+((method_signature
+  (method_identifier) @constructor)
+  (#any-of? @constructor "<init>" "<clinit>"))
 
 ; Fields
 
 (field_identifier) @field
 
-; Parameters
-
-(parameter) @parameter
+(annotation_key) @field
 
 ; Variables
 
-(variable) @variable
+(variable) @variable.builtin
+
+(local_directive
+  (identifier) @variable)
+
+; Parameters
+
+(parameter) @parameter ; more like @parameter.builtin but that doesn't exist in nvim-treesitter
```

Comment:
You can still use `@parameter.builtin` if you want.

Reply:
that's not in the list in CONTRIBUTING.md though, is that fine then?

Reply:
Yes. Only top-level captures are enforced; subcaptures can be arbitrary.

Reply:
Yes, it just won't be highlighted any differently by default.

Reply:
(careful: we just care about capture name to _allow_ consistent highlighting -- _how_ or even _whether_ something is highlighted completely depends on the colorscheme. we should be careful to always make this distinction.)